lib/setopt.c

```diff
@@ -2246,6 +2245,7 @@ CURLcode Curl_vsetopt(struct Curl_easy *data, CURLoption option,
                             va_arg(param, char *));
     break;
 
+#ifdef HAVE_LIBSSH2_KNOWNHOST_API
```

Comment:
libssh seems to use the `CURLOPT_SSH_KEYFUNCTION` etc as well so this is also slightly wrong. I think you can just remove the `#ifdef HAVE_LIBSSH2_KNOWNHOST_API`completely from the setopt logic. It seems unsuitable there.